# bioprograms

# CalcProteinMass
Calculate Protein mass with Amino acid weight dictionary

# CalculatingExpectedOffspring
Calculating Expected Offspring with Dominant Genotype

# CodonToProtein
Translate codon to amino acid(AA) using Codon to AA dictionary

# ConsensusSeqandProfile
Given: A collection of at most 10 DNA strings of equal length (at most 1 kbp) in FASTA format.
Return: A consensus string and profile matrix for the collection. 

# CountingKmersOfSequence
Counting Kmers Of Sequence

# CountingPointMutations
The Hamming distance between two strings having the same length is the minimum number of symbol substitutions
required to transform one string into the other. If the strings are given by s1 and s2, then we write the Hamming
distance between them as dH(s1,s2).

# FastaParseToDict
Parsing Fasta file(similar) to dictionary

# FibonacciSequence
Creating Fibonacci Sequences

# FibonaciRabbits
Given: Positive integers n≤40 and k≤5.
Return: The total number of rabbit pairs that will be present after n months, if we begin with 1 pair and in each
generation, every pair of reproduction-age rabbits produces a litter of k rabbit pairs (instead of only 1 pair).

# Generating_Kmers
Generating Kmers

# MendelianDominantInheritance
Given: Three positive integers k, m, and n, representing a population containing k+m+n organisms:
k individuals are homozygous dominant for a factor, m are heterozygous, and n are homozygous recessive.
Return: The probability that two randomly selected mating organisms will produce an individual possessing a
dominant allele (and thus displaying the dominant phenotype). Assume that any two organisms can mate.

# RNACodontoProtein
RNA Codon to amino acid dictionary

# SequenceDNAProteinRecordOOP
Sequence, DNA, Protein Record in OOP

# ORF_Finder
Using BioPython to pasrse through fasta file to get sequences, name
Find ORFs in different reading frames =1, 2, 3 

# StartPositionOfMotif
Finding the start positions of every motif

# StartPositionsOfKmer
Getting start positions of kmers of given length

# calGCcontentWithFasta
Computing GC content using FASTA files

# generateRandomDNAString
Generate Random DNA string

# longestCommonPrefix
Finding longest common prefix between 2 DNA strings

# matching2DNAstrings
To determine if 2 DNA strings are an exact match

# reverseComplement
Generate reverse complement from a DNA string
